### PR TITLE
Fix nrf5x's counter register to use 24 bits instead of 32 (fix #1515).

### LINKED ISSUE #1515: Invalid counter specification on nRF52(840)
According to the specification, the real-time counter only has 24 bits on nRF52840 ([source](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf52840%2Frtc.html&cp=4_0_0_5_21)). I could confirm this empirically on nRF52840-DK with [this example program](https://github.com/tock/libtock-rs/pull/120).

However, the current definition in the nrf5x chip mis-defines it as a 32-bit register.

https://github.com/tock/tock/blob/8039ff50976b2a1866cbd72f7fdef3982c0d8326/chips/nrf5x/src/rtc.rs#L43-L44

The `max_tics` is also incorrect and should be `(1 << 24) - 1` instead.

https://github.com/tock/tock/blob/8039ff50976b2a1866cbd72f7fdef3982c0d8326/chips/nrf5x/src/rtc.rs#L141-L143

However, I haven't seen any chip defining 24-bit alarms so far. This would be a first use case of https://github.com/tock/tock/pull/1413.

This means that in the current state, all bets are off regarding timers after 512 seconds (the counter being 32 kHz), as the overflows will be incorrectly computed.

This issue also relates to https://github.com/tock/tock/issues/1513.

### PR DESCRIPTION
**Pull request by gendx**
### Pull Request Overview

This pull request fixes the register definition of nrf5x's chips `counter` register in the RTC. According to the specification, this counter only contains 24 bits of value, and not 32.

This pull request only fixes the register definition, but the virtual alarm still need to be refactored to support 24-bit timestamps (see https://github.com/tock/tock/pull/1413). However, this pull request should keep the current behavior in the meantime, i.e. shouldn't break more things than are actually broken today.

### Testing Strategy

This pull request was tested by checking the [specification](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf52840%2Frtc.html&cp=4_0_0_5_21) and running [this program](https://github.com/tock/libtock-rs/pull/120) on an nRF52840-DK board to check that the observed counter is indeed 24 bits.

### TODO or Help Wanted

This pull request still needs support for 24-bit counters for all timers, alarms and virtual timers throughout the kernel. But I suggest to discuss that in https://github.com/tock/tock/pull/1413.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: chips/nrf5x/src/rtc.rs [hunk 1]
```diff
@@ -135,11 +135,11 @@ impl Time for Rtc<'a> {
     type Frequency = Freq32KHz;
 
     fn now(&self) -> u32 {
-        self.registers.counter.get()
+        self.registers.counter.read(Counter::VALUE)
     }
 
     fn max_tics(&self) -> u32 {
-        core::u32::MAX
+        (1 << 24) - 1
```
Comment (niklasad1) [inline chips/nrf5x/src/rtc.rs]: this could be a constant instead

Comment (gendx) [reply chips/nrf5x/src/rtc.rs]: I don't think it matters much, the compiler will optimize that to the constant anyway (and likely inline this function as well). Unless you think there should be a `u24::MAX` constant defined for the whole kernel?

Comment (niklasad1) [reply chips/nrf5x/src/rtc.rs]: I think you are right w.r.t optimizations but I was more concerned with readability for this

Comment (gendx) [reply chips/nrf5x/src/rtc.rs]: I think it would make sense to define the constant in the `Time` HIL as part of #1413.

Comment (niklasad1) [reply chips/nrf5x/src/rtc.rs]: yeah, let's merge this and revisit this converation there :)

**Comment by niklasad1** [comment]
bors +r

**Pull request merged**